main/performance.h

```diff
@@ -147,17 +140,35 @@ class Performance : public Object {
 	void set_physics_process_time(double p_pt);
 	void set_navigation_process_time(double p_pt);
 
-	void add_custom_monitor(const StringName &p_id, const Callable &p_callable, const Vector<Variant> &p_args);
+	void add_custom_monitor(const StringName &p_id, const Callable &p_callable, const Vector<Variant> &p_args, MonitorType p_type);
 	void remove_custom_monitor(const StringName &p_id);
 	bool has_custom_monitor(const StringName &p_id);
 	Variant get_custom_monitor(const StringName &p_id);
 	TypedArray<StringName> get_custom_monitor_names();
+	TypedArray<int> get_custom_monitor_types();
```

Comment:
I guess using a `TypedArray` here matches `get_custom_monitor_names()`, so this is probably fine. `Vector<int>` would be a somewhat faster and more compact data type for this